docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,508 @@
+      * `Passthrough`: traffic just passes a proxy without any modifications to the original destination, only available when defined sections `endpoints`, `tls` and `extension` are not defined.
+      * `Extension`: allows specifying a custom plugin for example, user can create a plugin which support AWS Lambda, when defined sections `endpoints` and `tls` are not available.
+    * **extension**: struct for a plugin configuration
+      * **type**: defines what kind of plugin to use, it's a string type so any new plugins should works.
+      * **config**: json map that is mapped to configuration provided in the type.
+    * **endpoints**: defines a list of endpoints
+      * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+      * **port**: defines a port of a destination.
+    * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
+      * **enabled**: defines if proxy should originate TLS.
+      * **version**: section for providing version specification.
+        * **min**: defines minmum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
+        * **max**: defines maximum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
+      * **allowRenegotiation**: defines if TLS sessions will allow renegotiation.
+      * **verification**: section for providing TLS verification details.
+        * **skipSAN**: defines if proxy should skip SAN verification. Default `false`.
+        * **subjectAltNames**: list of names to verify in the certificate.
+        * **caCert**: defines a certificate of CA.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientCert**: defines a certificate of a client.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientKey**: defines a client private key.
+          * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **ip**: allocated IP for a provided domain with `InternalVIP` type in a match section
+    * **type**: provides information about the way IP was provided
+  * **addresses**: list of domains and addresses
+    * **address**: IP address, CIDR, or domain name user
+    * **status**: indicate if an address is available
+    * **reason**: holds error messages if there are any 
+
+#### Cluster name
+
+Currently, we name clusters after their respective policies. However, when dealing with multiple protocols, we find it necessary to include the port number in the cluster name.
+Example:
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: httpbin
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  match:
+  - type: Domain
```

Comment:
what would happen if I configure two Domains with the same port but different protocol? Will this be caught during validation?

Reply:
We cannot detect all. There is no issue in the case of one policy, but if a user defines them in different MeshExternalServices we won't be able to detect them. We are going to create 2 entries with different policy name and matching. TLS and HTTP we can distinguish and create a proper matcher based on traffic.

Reply:
Because naming is important let's be clear: MeshExternalService is not a policy it's just a Resource.

Reply:
I'm resolving this - @Automaat - let us know if the answer was not sufficient.